sdk/search/Azure.Search.Documents/src/Pipeline/OdataMetadataPolicy.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Search.Documents.Pipeline
+{
+    /// <summary>
+    /// Sets the Accept HTTP header.
+    /// </summary>
+    internal class OdataMetadataPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly string _acceptValue;
+
+        private OdataMetadataPolicy(string level)
+        {
+            _acceptValue = "application/json; odata.metadata=" + level;
```

Comment:
Could we just set this on the swagger as a README transform and have it use the desired value per operation?  Then we don't have to worry about switching back and forth.

Reply:
You mean just have the generator use it unconditionally on operation classes? I can try it. All the tests will still need to be regenerated to include it.

Reply:
I spent too much time on this today and couldn't get anything that works reliably in a way that is robust to changes if the service team makes the same changes to swagger later. I'll stick with the policy here as other languages have already done.